compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -1627,23 +1678,28 @@ void generateNewXMLTextIns(BIRNonTerminator.NewXMLText newXMLText) {
 
     void generateNewXMLCommentIns(BIRNonTerminator.NewXMLComment newXMLComment) {
 
-        this.loadVar(newXMLComment.textOp.variableDcl);
-        this.mv.visitMethodInsn(INVOKESTATIC, XML_FACTORY, "createXMLComment",
-                String.format("(L%s;)L%s;", JvmPackageGen.IS_BSTRING ? JvmConstants.B_STRING_VALUE :
-                        STRING_VALUE, XML_VALUE),
-                false);
-        this.storeToVar(newXMLComment.lhsOp.variableDcl);
-    }
+            this.loadVar(newXMLComment.textOp.variableDcl);
+
+            loadType(this.mv, newXMLComment.type);
+
+            this.mv.visitMethodInsn(INVOKESTATIC, XML_FACTORY, "createXMLComment",
+                                    String.format("(L%s;L%s;)L%s;", JvmPackageGen.IS_BSTRING ?
+                                                          JvmConstants.B_STRING_VALUE : STRING_VALUE, BTYPE, XML_VALUE),
+                                    false);
+            this.storeToVar(newXMLComment.lhsOp.variableDcl);
+        }
 
     void generateNewXMLProcIns(BIRNonTerminator.NewXMLProcIns newXMLPI) {
 
-        this.loadVar(newXMLPI.targetOp.variableDcl);
-        this.loadVar(newXMLPI.dataOp.variableDcl);
-        String consVal = JvmPackageGen.IS_BSTRING ? JvmConstants.B_STRING_VALUE : STRING_VALUE;
-        this.mv.visitMethodInsn(INVOKESTATIC, XML_FACTORY, "createXMLProcessingInstruction",
-                String.format("(L%s;L%s;)L%s;", consVal, consVal, XML_VALUE), false);
-        this.storeToVar(newXMLPI.lhsOp.variableDcl);
-    }
+            this.loadVar(newXMLPI.targetOp.variableDcl);
+            this.loadVar(newXMLPI.dataOp.variableDcl);
+            loadType(this.mv, newXMLPI.type);
+
+            String consVal = JvmPackageGen.IS_BSTRING ? JvmConstants.B_STRING_VALUE : STRING_VALUE;
+            this.mv.visitMethodInsn(INVOKESTATIC, XML_FACTORY, "createXMLProcessingInstruction",
+                                    String.format("(L%s;L%s;L%s;)L%s;", consVal, consVal, BTYPE, XML_VALUE), false);
+            this.storeToVar(newXMLPI.lhsOp.variableDcl);
+        }
```

Comment:
looks like there is an alignment issue here. probably due to merge conflicts with jvm codegen package classes.